src/data/polynomial/algebra_map.lean

```diff
@@ -273,14 +285,21 @@ lemma aeval_eq_sum_range' [algebra R S] {p : R[X]} {n : ℕ} (hn : p.nat_degree
   aeval x p = ∑ i in finset.range n, p.coeff i • x ^ i :=
 by { simp_rw algebra.smul_def, exact eval₂_eq_sum_range' (algebra_map R S) hn x }
 
-lemma aeval_sum {ι : Type*} [algebra R S] (s : finset ι) (f : ι → R[X])
-  (g : S) : aeval g (∑ i in s, f i) = ∑ i in s, aeval g (f i) :=
-(polynomial.aeval g : R[X] →ₐ[_] _).map_sum f s
+lemma aeval_list_prod [algebra R S] (l : list R[X]) (x : S) :
+  aeval x l.prod = (l.map (aeval x)).prod :=
+eval₂_list_prod _ _ _
+
+lemma aeval_multiset_prod [algebra R S] (s : multiset R[X]) (x : S) :
+  aeval x s.prod = (s.map (aeval x)).prod :=
+eval₂_multiset_prod _ _ _
+
+lemma aeval_prod [algebra R S] {ι : Type*} (s : finset ι) (g : ι → R[X]) (x : S) :
+  aeval x (∏ i in s, g i) = ∏ i in s, aeval x (g i) :=
+eval₂_finset_prod _ _ _ _
 
-@[to_additive]
-lemma aeval_prod {ι : Type*} [algebra R S] (s : finset ι)
-  (f : ι → R[X]) (g : S) : aeval g (∏ i in s, f i) = ∏ i in s, aeval g (f i) :=
-(polynomial.aeval g : R[X] →ₐ[_] _).map_prod f s
+lemma eval_map_eq_aeval [algebra R S] (x : S) : eval x (p.map (algebra_map R S)) = aeval x p :=
+by simp_rw [eval_eq_sum_range' (nat.lt_add_one_iff.mpr (p.nat_degree_map_le _)),
```

Comment:
Is this `aeval_def` followed by `eval₂_eq_eval_map`? If so I'm not sure there's much point in this lemma

Reply:
At any rate, if you do keep it the proof should be something like
```suggestion
eval₂_eq_eval_map _
```

Reply:
Thanks! I just didn't realize it could be like this...